docs/reverse-proxy.rst

```diff
@@ -62,25 +57,59 @@ You can have the Waitress server use the ``https`` url scheme by default.:
 This works if all URLs generated by your application should use the ``https``
 scheme.
 
-Passing the ``X_FORWARDED_PROTO`` header to set ``wsgi.url_scheme``
--------------------------------------------------------------------
+Passing the proxy headers to setup the WSGI environment
+-------------------------------------------------------
 
 If your proxy accepts both HTTP and HTTPS URLs, and you want your application
-to generate the appropriate url based on the incoming scheme, also set up
-your proxy to send a ``X-Forwarded-Proto`` with the original URL scheme along
-with each proxied request.  For example, when using nginx::
+to generate the appropriate url based on the incoming scheme, you'll want to
+pass waitress ``X-Forwarded-Proto``, however Waitress is also able to update
+the environment using ``X-Forwarded-Proto``, ``X-Forwarded-For``,
+``X-Forwarded-Host``, and ``X-Forwarded-Port``::
+
+   proxy_set_header X-Forwarded-Proto $scheme;
+   proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+   proxy_set_header X-Forwarded-Host $host:$server_port;
+   proxy_set_header X-Forwarded-Port $server_port;
+
+when using Apache, ``mod_proxy`` automatically forwards the following headers::
 
-    proxy_set_header        X-Forwarded-Proto $scheme;
+   X-Forwarded-For
+   X-Forwarded-Host
+   X-Forwarded-Server
 
-or via Apache::
+You will also want to add to Apache::
 
    RequestHeader set X-Forwarded-Proto https
 
+Configure waitress's ``trusted_proxy_headers`` as appropriate::
+
+    trusted_proxy_headers = "x-forwarded-for, x-forwarded-host, x-forwarded-proto, x-forwarded-port"
+
+At this point waitress will set up the WSGI environment using the information
+specified in the trusted proxy headers. This will setup the following
+variables::
+
+   HTTP_HOST
+   SERVER_NAME
+   SERVER_PORT
+   REMOTE_ADDR
+   REMOTE_PORT (if available)
+   wsgi.url_scheme
+
+Waitress also has support for the `Forwarded (RFC7239) HTTP header
+<https://tools.ietf.org/html/rfc7239>`_ which is better defined than the ad-hoc
```

Comment:
use:

```
:rfc:`7239`
```

See example in https://docs.pylonsproject.org/projects/docs-style-guide/#rfcs